### PR TITLE
[Backport v2.7-branch] libc: minimal: add qsort to the minimal libc

### PR DESCRIPTION
**Pull request by zephyrbot**
Backport 60cc117c9c52f5b67ddd5a68fb210e5ed5471fc1~2..60cc117c9c52f5b67ddd5a68fb210e5ed5471fc1 from #39980

### CODE REVIEW

**Review by nordic-krch — APPROVED** [review]

**Review by stephanosio — APPROVED** [review]
Technically, this is a feature PR and we do not usually backport those.

Noting that this PR does not make any breaking changes, I have no objection to backporting this to v2.7; but, it would be good to get some reviews from the TSC and/or the release team.

**Comment by cfriedt** [comment]
> Technically, this is a feature PR and we do not usually backport those.
> 
> Noting that this PR does not make any breaking changes, I have no objection to backporting this to v2.7; but, it would be good to get some reviews from the TSC and/or the release team.

The topic was brought up in a release meeting a few weeks ago about whether it needed to go to the TSC. I think I captured it in the original PR, but the gist of that was - it's already being used in Zephyr for shell tests and is therefore less of a new feature and more of an improvement of an existing facility that is needed by testing (paraphrasing @nashif). I would be willing to bring it to the TSC if necessary though.

**Pull request merged**
